pkg/controller/migration/convert/network.go

```diff
@@ -41,30 +41,67 @@ func handleCalicoCNI(c *components, install *Installation) error {
 
 	errCtx := fmt.Sprintf("detected %s CNI plugin", plugin)
 
-	// CALICO_NETWORKING_BACKEND
-	netBackend, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "CALICO_NETWORKING_BACKEND")
-	if err != nil {
-		return err
-	}
-
-	if netBackend != nil && *netBackend != "" && *netBackend != "bird" {
-		return ErrIncompatibleCluster{fmt.Sprintf("%s, only CALICO_NETWORKING_BACKEND=bird is supported at this time", errCtx)}
-	}
-
-	// Calico CNI
 	if c.calicoCNIConfig == nil {
-		return ErrIncompatibleCluster{fmt.Sprintf("%s, required cni config was not found ", errCtx)}
+		return ErrIncompatibleCluster{fmt.Sprintf("%s, required Calico cni config was not found ", errCtx)}
 	}
 
 	if install.Spec.CNI == nil {
 		install.Spec.CNI = &operatorv1.CNISpec{}
 	}
 	install.Spec.CNI.Type = plugin
 
+	if install.Spec.CNI.IPAM == nil {
+		install.Spec.CNI.IPAM = &operatorv1.IPAMSpec{}
+	}
+
 	if install.Spec.CalicoNetwork == nil {
 		install.Spec.CalicoNetwork = &operatorv1.CalicoNetworkSpec{}
 	}
 
+	// CALICO_NETWORKING_BACKEND
+	netBackend, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "CALICO_NETWORKING_BACKEND")
+	if err != nil {
+		return err
+	}
+
+	switch c.calicoCNIConfig.IPAM.Type {
+	case "calico-ipam":
+		install.Spec.CNI.IPAM.Type = operatorv1.IPAMPluginCalico
+		install.Spec.CalicoNetwork.BGP = operatorv1.BGPOptionPtr(operatorv1.BGPEnabled)
+		switch {
+		case netBackend == nil:
+		case *netBackend == "":
+		case *netBackend == "bird":
+		case *netBackend == "vxlan":
+			install.Spec.CalicoNetwork.BGP = operatorv1.BGPOptionPtr(operatorv1.BGPDisabled)
+		default:
+			return ErrIncompatibleCluster{fmt.Sprintf("%s and IPAM plugin %s, CALICO_NETWORKING_BACKEND %s is not valid",
```

Comment:
Thinking out loud, does this belong in `validation.go`?

Reply:
I don't think we'd have a way to represent the config in the Installation API, so I don't think it would be possible to just migrate the config and have validation pick up that something was wrong.